### PR TITLE
make refinement provider implicit

### PR DESCRIPTION
**Pull request by lewisjkl**
Updates refinement provider to be provided to the refined schema as an implicit.

### CODE REVIEW

File: modules/codegen/src/smithy4s/codegen/Renderer.scala [hunk 1]
```diff
@@ -671,8 +671,15 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
...
-        line"${underlyingTpe.schemaRef}.refined(${renderNativeHint(hint)})($providerFqn)"
+      case Type.ExternalType(
+            _,
+            fqn,
+            maybeProviderImport,
+            underlyingTpe,
+            hint
+          ) =>
+        line"${underlyingTpe.schemaRef}.refined(${renderNativeHint(hint)})"
+          .addImport(maybeProviderImport.getOrElse(s"$fqn._"))
```
Comment (lewisjkl) [inline modules/codegen/src/smithy4s/codegen/Renderer.scala:682]: If I don't add the import for the companion object of the `targetType` explicitly, I get a diverging implicit expansion. I think scala expects the implicit to be in the companion object of the constraint by default and doesn't check the target type companion object by default?

File: modules/example/src/smithy4s/example/FancyList.scala [hunk 2]
```diff
@@ -1,11 +1,12 @@
 package smithy4s.example
 
 import smithy4s.Newtype
+import smithy4s.example.refined.FancyList._
```
Comment (lewisjkl) [inline modules/example/src/smithy4s/example/FancyList.scala:4]: Example of the import I am referring to in my previous comment

Comment (Baccata) [reply modules/example/src/smithy4s/example/FancyList.scala]: See [2ebcd04](https://github.com/disneystreaming/smithy4s/pull/346/commits/2ebcd046176466ced908069e4287221546f371a0)

File: modules/codegen/src/smithy4s/codegen/Renderer.scala (hunk 1 above)
Comment (kubukoz) [reply modules/codegen/src/smithy4s/codegen/Renderer.scala]: It might be a matter of annotating some types to get that type ineference to kick in

Comment (lewisjkl) [reply modules/codegen/src/smithy4s/codegen/Renderer.scala]: Yeah, Oli fixed it actually. See: https://github.com/disneystreaming/smithy4s/pull/346#discussion_r934938553

**Review by Baccata — APPROVED** [review]

**Pull request merged**